container_ci_suite/engines/container.py

```diff
@@ -21,69 +21,508 @@
+)
+
+logging.basicConfig(format="%(levelname)s:%(message)s", level=logging.DEBUG)
+logger = logging.getLogger(__name__)
 
-    @staticmethod
-    def docker_image_exists(image_name: str) -> bool:
-        """
-        Check if docker image exists or not
-        :param image_name: image to check
-        :return True: In case if image is present
-                False: In case if image is not present
-        """
-        output = PodmanCLIWrapper.run_docker_command(
-            f"images -q {image_name}", ignore_error=True, return_output=True)
-        return True if output != "" else False
 
-    @staticmethod
-    def docker_inspect(field: str, src_image: str) -> str:
+class ContainerImage(object):
+    def __init__(self, image_name: str):
+        self.image_name: str = image_name
+        self.container_args: str = ""
+        self.cid_file: Path = None
+        self.cid_file_dir: Path = None
+        logger.info(f"Image name to test: {image_name}")
+
+    # Replacement for ct_s2i_usage
+    def s2i_usage(self) -> str:
         return PodmanCLIWrapper.run_docker_command(
-            f"docker inspect -f '{field}' {src_image}"
+            f"run --rm {self.image_name} bash -c /usr/libexec/s2i/usage"
         )
 
-    @staticmethod
-    def docker_run_command(cmd):
-        return PodmanCLIWrapper.run_docker_command(f"run {cmd}")
+    def podman_run_cmd(self, cmd: str):
+        cmd_with_podman = cmd.replace("podman", "").replace("docker", "")
+        return PodmanCLIWrapper.run_docker_command(cmd=cmd_with_podman)
+
+    def is_image_available(self):
+        return PodmanCLIWrapper.run_docker_command(f"inspect {self.image_name}")
+
+    # Replacement for ct_container_running
+    def is_container_running(self):
+        return PodmanCLIWrapper.run_docker_command(f"inspect {self.image_name} -f '{{.State.Running}}'")
+
+    # Replacement for ct_container_exists
+    def is_container_exists(self, id_hash: str):
+        return PodmanCLIWrapper.run_docker_command(f"ps -q -a -f 'id={id_hash}'")
 
-    @staticmethod
-    def docker_get_user_id(src_image, user):
-        return PodmanCLIWrapper.docker_run_command(
-            f"--rm {src_image} bash -c 'id -u {user} 2>/dev/null"
+    # Replacement for ct_s2i_build_as_df
+    def s2i_build_as_df(self, app_path: str, s2i_args: str, src_image: str, dst_image: str):
+        named_tmp_dir = mkdtemp()
+        tmp_dir = Path(named_tmp_dir)
+        if tmp_dir.exists():
+            logger.debug("Temporary Directory exists.")
+        else:
+            logger.debug("Temporary directory not exists.")
+        ntf = mktemp(dir=str(tmp_dir), prefix="Dockerfile.")
+        df_name = Path(ntf)
+        df_content = self.s2i_create_df(
+            tmp_dir=tmp_dir,
+            app_path=app_path,
+            s2i_args=s2i_args,
+            src_image=src_image,
+            dst_image=dst_image,
         )
+        with open(df_name, mode="w") as f:
+            f.writelines(df_content)
```

Comment:
https://docs.python.org/3/library/io.html#io.IOBase.writelines
'Line separators are not added, so it is usual for each of the lines provided to have a line separator at the end.'

Not adding `\n` to every line in `df_content` will result in the 'Error: FROM requires either one argument, or three: FROM <source> [AS <name>]' - the resulting dockerfile will have one single line only.

Reply:
Function `s2i_create_df` returns list of `List[str]`. But I will re-write it a bit. 
Commit https://github.com/sclorg/container-ci-suite/pull/93/commits/84e5eb6caa17b7ba2eea078a8292268b46efa952